hazelcast/src/main/java/com/hazelcast/query/impl/predicates/LikePredicate.java

```diff
@@ -43,6 +47,38 @@ public LikePredicate(String attributeName, String expression) {
         this.expression = expression;
     }
 
+    @Override
+    public Set<QueryableEntry> filter(QueryContext queryContext) {
+        Index index = queryContext.matchIndex(attributeName, QueryContext.IndexMatchHint.PREFER_ORDERED);
+        String indexPrefix = expression.substring(0, expression.length() - 1);
```

Comment:
As I understand we cut the `%` here. But what if we have trailing whitespace? Is it possible?

Reply:
if `%` is not the last symbol in the expression it wouldn't be indexed, added tests https://github.com/hazelcast/hazelcast/pull/18289/files#diff-8b3c97074f4277303b021451ae7612b8de726a0aec157799b64d5673798b363aR144